### PR TITLE
[Alaukik Pant] iP

### PR DESCRIPTION
**Pull request by alaukiknpant**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -1,10 +1,137 @@
...
+
+
+    //Add a task
+    public void addTask(String line, ArrayList<Task> taskList) {
+
+        String feedIn = line.toUpperCase();
+        String ogString = line;
+        while (!feedIn.equals("BYE")) {
+            if (markDone(ogString, taskList)){
+                System.out.println("___________________________________________________________________________\n");
```
Comment (yuchenlichuck) [inline src/main/java/Duke.java]: There are lots of repeat statements. It'd better extract these statements to a function.

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,137 @@
...
+
+
+        }
+        System.out.println("Bye. Hope to see you again soon!\n");
+        System.out.println("___________________________________________________________________________");
+
+    }
+
+
+    //Get things started
```
Comment (yuchenlichuck) [inline src/main/java/Duke.java]: it'd better change the comments from `//` to `/**`

File: src/main/java/Task.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public String getDescription() {
+
```
Comment (yuchenlichuck) [inline src/main/java/Task.java]: It'd better no space.

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,137 @@
...
+                System.out.println("___________________________________________________________________________");
+                Task t = new Task(ogString);
+                taskList.add(t);
+                System.out.println("added: " + ogString + "\n");
+                System.out.println("___________________________________________________________________________");
+            }
+            Scanner scanner = new Scanner(System.in);
+            ogString = scanner.nextLine();
+            feedIn= ogString.toUpperCase();
+
```
Comment (yuchenlichuck) [inline src/main/java/Duke.java]: It's better no double blank lines in the end.

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,137 @@
...
+    }
+
+    //Print all available tasks
+    public void printTaskList(ArrayList<Task> taskList){
+        int i = 1;
+        for (Task task : taskList){
+            String description = task.getDescription();
+            String statusIcon = task.getStatusIcon();
+            System.out.println(i + ". "  + "[" +  statusIcon + "]" + " " + description);
+            i+= 1;
```
Comment (yuchenlichuck) [inline src/main/java/Duke.java]: use `i++` frequently to replace` i+= 1`

**Review by yuchenlichuck — COMMENTED** [review]
Well done! Good job! I think the blank lines can be decreased. And some statements used repeat can be abstract to a method to save letters.

Review together with @JTWang2000

File: src/main/java/Duke.java (hunk 1 above)
Comment (alaukiknpant) [reply src/main/java/Duke.java]: I agree!

File: src/main/java/Duke.java (hunk 2 above)
Comment (alaukiknpant) [reply src/main/java/Duke.java]: Yes, agree!

File: src/main/java/Task.java (hunk 3 above)
Comment (alaukiknpant) [reply src/main/java/Task.java]: Yes, true, thanks

File: src/main/java/Duke.java (hunk 4 above)
Comment (alaukiknpant) [reply src/main/java/Duke.java]: Agreed

File: src/main/java/Duke.java (hunk 5 above)
Comment (alaukiknpant) [reply src/main/java/Duke.java]: yes, agreed

**Pull request closed without merging**